### DISCUSSION TITLE
fastify and ESLint - Ajv version conflict

### DISCUSSION THREAD

**Discussion by hamanuha** [fastify/fastify#4837] [Q&A]
Hey guys,

I am trying to validate a date-time property that requires a minimum date.
But I can't get this to work. I always get this error:

`{(intermediate value)(intermediate value)(intermediate value)(intermediate value)}.compare is not a function`

This the part of the schema I use for the date time:
```typescript
properties: {
    datetime: { type: 'string', format: 'date-time', formatMinimum: '2023-01-01T00:00:00.000' },
```

And this is the value that is being sent:
`datetime: '2023-06-20T06:26:00.000Z'`

I tried setting up ajv-formats by myself, but it doesn't change anything.
Can you help me here?

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Reply by hamanuha** [reply]
I created a minimal example with only fastify and this seems to work. After some investigation I found out the issue pops up when using fastify together with ESLint. ESLint is using an older version of ajv (6.12.6).
Any ideas on how to solve this?

Example:

**package.json**
```json
{
  "type": "module",
  "dependencies": {
    "fastify": "4.18.0",
    "eslint": "8.43.0"
  }
}
```

**server.js**
```javascript
import fastify from 'fastify'

const server = fastify()

server.route({
  method: 'POST',
  url: '/',
  schema: {
    body: {
      type: 'object',
      required: ['datetime'],
      properties: {
        datetime: { type: 'string', format: 'date-time', formatMinimum: '2023-01-01T00:00:00.000' },
      },
      additionalProperties: false,
    },
  },
  handler: async (request, reply) => {
    return request.body.datetime
  },
})

const start = async () => {
  try {
    await server.listen({ port: 3001 })
  } catch (err) {
    server.log.error(err)
    process.exit(1)
  }
}
start()
```

Run the server
`node server`

Call the route
`curl -X POST localhost:3001 -H "Content-Type: application/json" -d '{"datetime": "2022-06-20T06:30:00.000Z"}'`

**Comment by hamanuha** [comment]
Installing Ajv as dependency (although not directly used) solves the problem. Very strange.
